CrazyCanvas/Source/GUI/SavedServerGUI.cpp

```diff
@@ -10,57 +10,71 @@ using namespace LambdaEngine;
 using namespace Noesis;
 
 
-SavedServerGUI::SavedServerGUI(const LambdaEngine::String& xamlFile) :
-	m_ItemCount(0)
+SavedServerGUI::SavedServerGUI(const LambdaEngine::String& xamlFile)
 {
 
 }
 
-void SavedServerGUI::Init(ListBox* pListView)
+void SavedServerGUI::Init(ListBox* pSavedListView, ListBox* pLocalListView)
 {
-	/*textBlock = *new ObservableCollection<TextBlock>();
-	tBlock = *new TextBlock("ChrilleBoi");
-	textBlock->Add(tBlock);
-	m_pSavedServerList->SetItemsSource(textBlock);*/
-	
-	m_SavedServerList = *pListView;
-
+	m_SavedServerList = *pSavedListView;
+	m_LocalServerList = *pLocalListView;
 }
 
 SavedServerGUI::~SavedServerGUI()
 {
 }
 
-void SavedServerGUI::AddServerItem(Grid* pParentGrid, const char* pServerN, const char* pMapN, const char* pPing, bool isRunning)
+Ptr<Grid> SavedServerGUI::AddSavedServerItem(Grid* pParentGrid, const ServerInfo& serverInfo, bool isRunning)
+{
+	Ptr<Grid> grid = AddServerItem(pParentGrid, serverInfo, isRunning);
+	if (m_SavedServerList->GetItems()->Add(grid) == -1)
+	{
+		LOG_ERROR("Refreshing Saved List");
+	}
+
+	return grid;
+}
+
+Ptr<Grid> SavedServerGUI::AddLocalServerItem(Grid* pParentGrid, const ServerInfo& serverInfo, bool isRunning)
+{
+	Ptr<Grid> grid = AddServerItem(pParentGrid, serverInfo, isRunning);
+	if (m_LocalServerList->GetItems()->Add(grid) == -1)
+	{
+		LOG_ERROR("Refreshing Local List");
+	}
+
+	return grid;
+}
+
+Ptr<Grid> SavedServerGUI::AddServerItem(Grid* pParentGrid, const ServerInfo& serverInfo, bool isRunning)
 {
-	
 	Ptr<Grid> grid = *new Grid();
 
-	for (int i = 0; i < 4; i++)
+	for (int i = 0; i < 5; i++)
```

Comment:
Could not the hardcoded loop limit be extracted from a length of something?
Not sure why 5 is the correct number.

Reply:
As you can see the index i is never used and it does therefore not iterate over anything. Therefore a constant of 5 is perfectly valid

Reply:
The grid doesn't know how many children it has yet, but I'll see if there's a way to get the number of columns.